src/H5FDlog.c

```diff
@@ -321,19 +321,17 @@ H5FD__log_term(void)
 herr_t
 H5Pset_fapl_log(hid_t fapl_id, const char *logfile, unsigned long long flags, size_t buf_size)
 {
-    H5FD_log_fapl_t fa;        /* File access property list information */
-    H5P_genplist_t *plist;     /* Property list pointer */
-    herr_t          ret_value; /* Return value */
+    H5FD_log_fapl_t fa;           /* File access property list information */
+    H5P_genplist_t *plist = NULL; /* Property list pointer */
+    herr_t          ret_value;    /* Return value */
 
     FUNC_ENTER_API(FAIL)
     H5TRACE4("e", "i*sULz", fapl_id, logfile, flags, buf_size);
 
-    /* Do this first, so that we don't try to free a wild pointer if
-     * H5P_object_verify() fails.
-     */
+    /* Initialize. */
```

Comment:
It's important to do this before calling something that can jump to the error condition. If someone moves this down, the bug will resurface. Also, "wild pointer" is a known term and the pointer in question isn't the plist pointer, it's a char pointer in the H5FD_log_fapl_t struct.

https://en.wikipedia.org/wiki/Dangling_pointer

Reply:
Also, we typically don't add periods at the end of one-line coments.